rest/config.go

```diff
@@ -794,48 +785,53 @@ func (self *ServerConfig) MergeWith(other *ServerConfig) error {
 	for _, flag := range other.DeprecatedLog {
 		self.DeprecatedLog = append(self.DeprecatedLog, flag)
 	}
+	if self.Logging == nil {
+		self.Logging = other.Logging
+	}
 	if other.Pretty {
 		self.Pretty = true
 	}
 	for name, db := range other.Databases {
 		if self.Databases[name] != nil {
 			return base.RedactErrorf("Database %q already specified earlier", base.UD(name))
 		}
+		if self.Databases == nil {
+			self.Databases = make(DbConfigMap)
+		}
 		self.Databases[name] = db
 	}
 	return nil
 }
 
 // Reads the command line flags and the optional config file.
-func ParseCommandLine() (err error) {
-	addr := flag.String("interface", DefaultInterface, "Address to bind to")
-	authAddr := flag.String("adminInterface", DefaultAdminInterface, "Address to bind admin interface to")
-	profAddr := flag.String("profileInterface", "", "Address to bind profile interface to")
-	configServer := flag.String("configServer", "", "URL of server that can return database configs")
-	deploymentID := flag.String("deploymentID", "", "Customer/project identifier for stats reporting")
-	couchbaseURL := flag.String("url", DefaultServer, "Address of Couchbase server")
-	poolName := flag.String("pool", DefaultPool, "Name of pool")
-	bucketName := flag.String("bucket", "sync_gateway", "Name of bucket")
-	dbName := flag.String("dbname", "", "Name of Couchbase Server database (defaults to name of bucket)")
-	pretty := flag.Bool("pretty", false, "Pretty-print JSON responses")
-	verbose := flag.Bool("verbose", false, "Log more info about requests")
-	logKeys := flag.String("log", "", "Log keys, comma separated")
-	logFilePath := flag.String("logFilePath", "", "Path to log files")
-	certpath := flag.String("certpath", "", "Client certificate path")
-	cacertpath := flag.String("cacertpath", "", "Root CA certificate path")
-	keypath := flag.String("keypath", "", "Client certificate key path")
-
+func ParseCommandLine(args []string, handling flag.ErrorHandling) (err error) {
```

Comment:
As discussed, change to return *ServerConfig and error, and then have the caller in ServerMain assign that to the global 'config' var.  (allows tests to not interact with the global var)

Reply:
This is also included with the latest revision.